File: app/src/main/java/djordjeh/architecture/mvp/ui/task/ContractTask.java
package djordjeh.architecture.mvp.ui.task;


import androidx.annotation.NonNull;

import djordjeh.architecture.mvp.data.model.Task;
import djordjeh.architecture.mvp.ui.BasePresenter;

public interface ContractTask {

    interface View {
        void showTask(@NonNull Task task);
        void showEmptyTitleError(boolean show);
        void onTaskSaved(Task task);
        void showError(Throwable throwable);
    }

    interface Presenter extends BasePresenter {
        void getTask(long taskId);
        void saveTask(Task task);
    }
}

File: app/src/main/java/djordjeh/architecture/mvp/ui/task/ContractTask.kt
package djordjeh.architecture.mvp.ui.task


import djordjeh.architecture.mvp.data.model.Task
import djordjeh.architecture.mvp.ui.BasePresenter

interface ContractTask {

    interface View {
        fun showTask(task: Task)
        fun showEmptyTitleError(show: Boolean)
        fun onTaskSaved(task: Task)
        fun showError(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
        fun getTask(taskId: Long)
        fun saveTask(task: Task)
    }
}
